### ISSUE TITLE
FOF-/F0F Code Templates

### ISSUE THREAD

**Issue by peterpeter** [elkuku/EasyCreator#32]
Hi Nicolas
Related to #30 (EC still alive?) some toughts of mine.
I'm a developer in a web-agency, writing smaller client specific extensions (up to 20+ per year). I am so to say an ideal candidate for heavingly using Easy-Creator. And that was the case in the 'J! 1.5x Decade'. In the J!-2.5's, I wrote just a few extensions, without EC. So, now in the J!-3.x.x's, I discovered the FOF-(F0F-)Framework, and it's advantages: writing components with just a few lines code. Over 90% is automaticly computed and rendered by the framework itself (toolbar, backend- and frontend-views, filtering, multilanguage, ACL.....). Including inbuild back- and future compatibility! 

And I'm doing that with PHPStorm-file-templates, made by myself, and it works like a charm. Just if you need a special task or view (like a file-upload or xml-export) you need to write code. The CRUD of simple record's are fully inbuild.

What about if we lift Easy-Creator to the next (evolution?) level by combine the advantages of both, FOF and EC?
By providing a FOF-Component Template(s)/Skeleton?

I've been play around a little to create my own EC-component-templates, some times ago, and it seems not to be too complicate, as EC is easy templates-extensible, by it's very smart and clever code design ;-)

Or do you either want to stick to J!'s native (component-) code? Oder better spend some effort in that wished/planned PHPStorm Joomla-Plugin?  https://youtrack.jetbrains.com/issue/WI-22979 

Would be glad, to here some thoughts about that from you. :)

**Comment by HermanPeeren** [comment]
You could do a pull request to this repo to share your FOF-templates with us. I'm working on some Nooku Framework 2.0 templates for EC and will do a PR too. One of the strong points of EC is that you can make your own templates, so it can be AND-AND instead of OR-OR.

BTW: FOF was inspired by Nooku, which was too volatile at that moment. Although FOF has evolved a bit and has some nice things of its own (like FofForm), Nooku Framework has new basic classes instead of building on old Joomla stuff. I especaally like heir Chain fo Command, Mixins and RESTful interface. I am not affiliated to them, just enthusiastic. http://www.nooku.org/

**Comment by peterpeter** [comment]
Yes Herman, of course, you are right:

Build it and make a PR, Sharing is caring ;-)
Just wanna hear some other toughts about that.
I'll give it a try, as soon I have worked some of my other tasks.

Yes, I know the storys behind both, Nooku and FOF/F0F.
But for me and my daily extension requirements FOF fits perfectly: Fast development, DRY, and It works perfect for me and for now.

Sure, rather less design patterns, but more practicable on the other way (as I said: for me and for now).

**Comment by HermanPeeren** [comment]
I'm very interested in your FOF-templating attempts and we can also help each other with it! It would be great if more templates would be available for EC. So I will keep you informed about my attempts for Nooku scaffolding with EC.

**Comment by elkuku** [comment]
Yes I would love all kind of extension templates to be included in EasyCreator :wink: 

So, I created a repository where all EasyCreator templates will live: https://github.com/EasyCreator/templates

Those templates are installable from within EasyCreator.

Pull requests are welcome :wink:
